pipeline.cpp

```diff
@@ -73,17 +73,68 @@ void DrawLine(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* cons
 {
     // Your code goes here
 }
-
+/********************************************
+ *Determinant
+ *Find the area of the triangle between
+ * two vectors. 
+ ******************************************/
+double Determinant(Vertex v1, Vertex v2) 
+{
+    return (v1.x * v2.y) - (v1.y * v2.x); 
+}
 /*************************************************************
  * DRAW_TRIANGLE
  * Renders a triangle to the target buffer. Essential 
  * building block for most of drawing.
  ************************************************************/
 void DrawTriangle(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* const attrs, Attributes* const uniforms, FragmentShader* const frag)
 {
-    // Your code goes here
+//get the box around the triangle
+float maxX = MAX3(triangle[0].x,triangle[1].x,triangle[2].x);
+float maxY = MAX3(triangle[0].y,triangle[1].y,triangle[2].y);
+float minX = MIN3(triangle[0].x,triangle[1].x,triangle[2].x);
+float minY = MIN3(triangle[0].y,triangle[1].y,triangle[2].y);
+
+Vertex vt1;
+Vertex vt2;
+
+vt1.x = (triangle[1].x - triangle[0].x);
```

Comment:
Minor: All these variables can be initialized when creating the new object using <code>Vertex vt1 = {assignments;};</code>